pandas/_libs/src/vendored/ujson/python/objToJSON.c

```diff
@@ -373,6 +373,31 @@ static char *PyTimeToJSON(JSOBJ _obj, JSONTypeContext *tc, size_t *outLen) {
   return outValue;
 }
 
+static char *PyDecimalToUTF8Callback(JSOBJ _obj, JSONTypeContext *tc,
+                                     size_t *len) {
+  PyObject *obj = (PyObject *)_obj;
+  PyObject *str = PyObject_Str(obj);
+  if (str == NULL) {
+    *len = 0;
+    if (!PyErr_Occurred()) {
```

Comment:
Hmm doesn't Python always guarantee to see the global error when `PyObject_Str` returns `NULL`? I think a lot of this can be simplified by just returning `NULL` in case of an error (after DECREF'ing `str`)